### PR TITLE
feat: added internationalization features to the dashboard.

### PR DESCRIPTION
**Pull request by GaKireiTsuki**
1. If you need to add new translations, please create a new JSON file under `apps/dokploy/i18n/locales` (use the lowercase value of `defaultLanguageTag` from `apps/dokploy/i18n/storefronts.ts` as the filename).
2. Uncomment the corresponding item in `storefronts.ts`.
3. After that, restart the development server.
4. Select your language under `/dashboard/settings/appearance` to ensure the page displays the language you selected, rather than the fallback language.
5. Once everything is confirmed to be working correctly, you can submit your Pull Requests."

https://github.com/user-attachments/assets/a8c52b46-1eff-4f95-a1d8-8a90dde11fce

localhost_3000_
localhost_3000_ (1)

### CODE REVIEW

**Comment by SashaGoncharov19** [comment]
Bump, waiting for approval.
I want to add Ukrainian localisation

**Comment by Siumauricio** [comment]
@GaKireiTsuki can you add a language switcher in the `/dashboard/settings/appearance` route?
image

**Comment by GaKireiTsuki** [comment]
@Siumauricio review

File: apps/dokploy/next.config.mjs
Comment (Siumauricio) [inline apps/dokploy/next.config.mjs]: I pushed a feature tag to dockerhub to test this and I'm getting this error which I suspect is coming from this
image

File: apps/dokploy/i18n/InternationalizationKit/smartLocale/smartLocale.js
Comment (Siumauricio) [inline apps/dokploy/i18n/InternationalizationKit/smartLocale/smartLocale.js]: apply all the biome linter suggestions

image

image

File: apps/dokploy/i18n/InternationalizationKit/interface/locales.d.ts
Comment (Siumauricio) [inline apps/dokploy/i18n/InternationalizationKit/interface/locales.d.ts]: How did you generate this file? do you have command?

**Review by Siumauricio — CHANGES_REQUESTED** [review]
I don't understand why it the bundle size increases 250mb is alot
image

**Comment by GaKireiTsuki** [comment]
> I don't understand why it the bundle size increases 250mb is alot and I don't see you are including any library which is strange image

I think it's because of the JavaScript `Intl` object.

File: apps/dokploy/i18n/InternationalizationKit/interface/locales.d.ts
Comment (GaKireiTsuki) [reply apps/dokploy/i18n/InternationalizationKit/interface/locales.d.ts]: > How did you generate this file? do you have command?

It is necessary to use `smartLocale.js` to generate `locales.d.ts`, and configure `SmartLocalePlugin` in `next.config.mjs`.

```javascript
		config.plugins.push(new SmartLocalePlugin({
			inputDirectory: "./i18n/locales/",
			outputDirectory: "./i18n/InternationalizationKit/interface/",
			outputImportDirectory: "./i18n/",
			defaultLocale: "en-us",
			fileType: "json",
		}));

Comment (Siumauricio) [reply apps/dokploy/i18n/InternationalizationKit/interface/locales.d.ts]: So they generate automatically?

Comment (GaKireiTsuki) [reply apps/dokploy/i18n/InternationalizationKit/interface/locales.d.ts]: That's right, it will execute once when the development server starts.

**Comment by PaiJi** [comment]
Are you planning to create a zh translation for the website as well? i'm currently working on this. if you are also doing it, i can pause my efforts.

**Comment by Siumauricio** [comment]
Have you checked this error? @GaKireiTsuki 

https://github.com/Dokploy/dokploy/pull/409#discussion_r1744808871

**Comment by Siumauricio** [comment]
Looks like the active state is not working on settings
image

**Comment by GaKireiTsuki** [comment]
> 看起来活动状态在设置上不起作用 image

this issue has been fixed.

**Comment by GaKireiTsuki** [comment]
> Have you checked this error? @GaKireiTsuki
> 
> [#409 (comment)](https://github.com/Dokploy/dokploy/pull/409#discussion_r1744808871)

I set the SmartLocalePlugin to run when config.mode === 'development', which may help improve this issue.

**Comment by Siumauricio** [comment]
Same error @GaKireiTsuki 
image

**Comment by PaiJi** [comment]
Hi, i had add basic i18n support, can you help translation rest of zh-Hans key?

**Comment by GaKireiTsuki** [comment]
Sorry, I don't have much time for translation work at the moment.> Hi, i had add basic i18n support, can you help translation rest of zh-Hans key?

**Pull request closed without merging**
